drivers/mipi_dbi/mipi_dbi_spi.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_mipi_dbi_spi
+
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/gpio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mipi_dbi_spi, CONFIG_MIPI_DBI_LOG_LEVEL);
+
+struct mipi_dbi_spi_config {
+	/* SPI hardware used to send data */
+	const struct device *spi_dev;
+	/* Command/Data gpio */
+	const struct gpio_dt_spec cmd_data;
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	/* Tracks if C/D is inverted from traditional polarity */
+	bool cd_invert;
+	/* Tracks if the reset signal is inverted from traditional polarity */
+	bool rst_invert;
```

Comment:
See my comment above- unfortunately I think we need to use these properties over the GPIO flags, since the pins won't always be GPIO signals on other IP blocks